feat: Add SAP HANA support

dazhuangc opened the pull request:
<!--
Thanks for wanting to fix something on Sequelize - we already love you!
Please fill in the template below.
If unsure about something, just do as best as you're able.
-->

## Pull Request Checklist

<!-- Please make sure to review and check all of these items: -->

- [x] Have you added new tests to prevent regressions?
- [x] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? <!-- Put PR link here --> https://github.com/sequelize/website/pull/799
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

<!-- NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open. -->

## Description of Changes

Added dialect support, data types, query generator in order to support SAP HANA natively from sequelize.

## List of Breaking Changes

No breaking changes.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added SAP HANA as a supported Sequelize database dialect.
  * Added the `@sequelize/hana` connector with HANA-specific connectivity, data types, transactions, schemas, indexes, and SQL generation.
  * Added local Docker-based HANA development and testing commands.
  * Added HANA unit and integration test coverage across supported Sequelize features.

* **Documentation**
  * Updated the README and contribution guide with HANA support, setup, testing, and example commands.

* **Chores**
  * Added HANA validation to continuous integration and release checks.
  * Enabled `hana` as a semantic pull request scope.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dazhuangc committed:
feat(hana)!: create hana dialect in the @sequelize/hana package

dazhuangc committed:
feat(hana)!: hana integration tests

dazhuangc committed:
feat(hana)!: HANA integration tests

HANA transaction management.

dazhuangc committed:
feat(hana)!: HANA integration tests

Create table if not exists.

dazhuangc committed:
feat(hana)!: HANA integration tests

BOOLEAN data type in INSERT statements.

dazhuangc committed:
feat(hana)!: HANA integration tests

attributeToSQL() in query-generator.js

dazhuangc committed:
feat(hana)!: HANA integration tests

Sequelize.literal

dazhuangc committed:
feat(hana)!: HANA integration tests

HANA TINYINT is unsigned by default.

dazhuangc committed:
feat(hana)!: HANA integration tests

UniqueConstraintError when model is not passed.

dazhuangc committed:
feat(hana)!: HANA integration tests

Return id on bulkInsert.

dazhuangc committed:
feat(hana)!: HANA integration tests

Fix test 'should not contain views'

dazhuangc committed:
feat(hana)!: HANA integration tests

Fix test rename a column.

dazhuangc committed:
feat(hana)!: HANA integration tests

Fix test remove a column.

dazhuangc committed:
feat(hana)!: HANA integration tests

PUBLIC is one of system defined schemas;
SYSTEM is not.

dazhuangc committed:
feat(hana)!: HANA integration tests

Use explicit afterTable.schema when renaming table.

dazhuangc committed:
feat(hana)!: HANA integration tests

Add badAddressConfig for HANA.

dazhuangc committed:
feat(hana)!: HANA integration tests

Add badUsernameConfig for HANA.

dazhuangc committed:
feat(hana)!: HANA integration tests

Add noPasswordConfig for HANA.

dazhuangc committed:
feat(hana)!: HANA integration tests

HANA does not support dropping primary key or foreign key columns.

dazhuangc committed:
feat(hana)!: HANA integration tests

changeColumn.test.js: DataTypes.FLOAT in HANA is REAL.

dazhuangc committed:
feat(hana)!: HANA integration tests

changeColumn.test.js: use explicit 'NULL' when changing from NOT NULL TO nullable

dazhuangc committed:
feat(hana)!: HANA integration tests

insert query:
skip BaseSqlExpression parameter names only when the value is not bound to a `$sequelize_${i}`

dazhuangc committed:
feat(hana)!: HANA integration tests

formatError ERR_SQL_FK_NOT_FOUND, code=461

dazhuangc committed:
feat(hana)!: HANA integration tests

add-show-remove-constraint.test.ts: constraints query and UnknownConstraintError handling

dazhuangc committed:
feat(hana)!: HANA integration tests

model.test.js: HANA does not have collate on index columns

dazhuangc committed:
feat(hana)!: HANA integration tests

add-show-remove-constraint.test.ts: fix updateAction

dazhuangc committed:
feat(hana)!: HANA integration tests

query-interface.test.js: failed to drop 'SYSTEM' schema

dazhuangc committed:
feat(hana)!: HANA integration tests

destroy-all.test.ts: HANA does not have 'NO ACTION' referential action

dazhuangc committed:
feat(hana)!: HANA integration tests

error.test.ts: ForeignKeyConstraintError handling

dazhuangc committed:
feat(hana)!: HANA integration tests

error.test.ts: unique constraint error handling

dazhuangc committed:
feat(hana)!: HANA integration tests

createTableQuery(): use column inline COMMENT clause

dazhuangc committed:
feat(hana)!: HANA integration tests

transaction.test.js: FROM DUMMY syntax

dazhuangc committed:
feat(hana)!: HANA integration tests

hana does not support deferrable

dazhuangc committed:
feat(hana)!: HANA integration tests

transaction.test.ts: use SQLSCRIPT_SYNC:SLEEP_SECONDS()

dazhuangc committed:
feat(hana)!: HANA integration tests

describeTable.test.js: describeTable()

dazhuangc committed:
feat(hana)!: HANA integration tests

add options parameter to queryInterface.tableExists()

dazhuangc committed:
feat(hana)!: HANA integration tests

query.js: support isRawQuery() in formatResults()

dazhuangc committed:
feat(hana)!: HANA integration tests

query.test.js: fix SELECT syntax

dazhuangc committed:
feat(hana)!: HANA integration tests

error.test.ts: unique constraint error handling

dazhuangc committed:
feat(hana)!: HANA integration tests

transaction.test.ts: hana does not support READ UNCOMMITTED

dazhuangc committed:
feat(hana)!: HANA integration tests

sequelize.test.js: fix syntax for HANA

dazhuangc committed:
feat(hana)!: HANA integration tests

simulate IF NOT EXISTS for createTable() and dropTable()

dazhuangc committed:
feat(hana)!: HANA integration tests

pool.test.ts: set connection id when initializing

dazhuangc committed:
feat(hana)!: HANA integration tests

connectionTransactionMethods: true
but hana use SQL statement to set isolation level

dazhuangc committed:
feat(hana)!: HANA integration tests

bulk-create.test.js: return supplied values on primary keys

dazhuangc committed:
feat(hana)!: HANA integration tests

transaction.test.ts: hana REPEATABLE READ and SERIALIZABLE

dazhuangc committed:
feat(hana)!: HANA integration tests

transaction.test.ts: use REPEATABLE_READ as initial isolation level

dazhuangc committed:
feat(hana)!: HANA integration tests

transaction.test.ts: use REPEATABLE_READ for nestMode "separate".
hana does not support READ_UNCOMMITTED.

dazhuangc committed:
feat(hana)!: HANA integration tests

create.test.js: skip tests with UniqueConstraintError.
hana does not return inserted values in error message yet.

dazhuangc committed:
feat(hana)!: HANA integration tests

error.test.ts: handle value in unique constraint violated error message

dazhuangc committed:
feat(hana)!: HANA integration tests

query-interface: wrap commit/rollback with try-finally

dazhuangc committed:
feat(hana)!: HANA integration tests

transaction.test.ts: import type

dazhuangc committed:
feat(hana)!: HANA integration tests

query.js: refactor run()

dazhuangc committed:
feat(hana)!: HANA integration tests

belongs-to-many.test.js: hana doesn't have a real SERIALIZABLE isolation mode

dazhuangc committed:
feat(hana)!: HANA unit tests

createSchemaQuery options

dazhuangc committed:
feat(hana)!: HANA integration tests

update saplabs/hanaexpress docker image to 2.00.076.00.20240701.1

dazhuangc committed:
feat(hana)!: HANA unit tests

query-generator.js: handle undefined modelAttributes and options

dazhuangc committed:
feat(hana)!: HANA integration tests

createTable.test.ts: [HANA Specific] create row table

dazhuangc committed:
feat(hana)!: HANA integration tests

create table with comment

dazhuangc committed:
feat(hana)!: HANA unit tests

decimal-numbers.test.ts: support unconstrained DECIMAL

dazhuangc committed:
feat(hana)!: HANA integration tests

HANA does not support collate for index

dazhuangc committed:
feat(hana)!: HANA unit tests

integers.test.ts: implement sequelize TINYINT with HANA integers.
HANA TINYINT is unsigned by default.

dazhuangc committed:
feat(hana)!: HANA unit tests

misc-data-types.test.ts: implement JSON with HANA NVARCHAR(5000)

dazhuangc committed:
feat(hana)!: HANA unit tests

bulk-delete-query.test.ts: bulk delete

dazhuangc committed:
feat(hana)!: HANA unit tests

commit-transaction-query.test.ts: commit transaction query
rollback-transaction-query.test.ts: rollback transaction query

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table-query.test.ts: ifNotExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table-query.test.ts: column comments

dazhuangc committed:
feat(hana)!: HANA unit tests

describe-table-query.test.ts: describeTableQuery()

dazhuangc committed:
feat(hana)!: HANA unit tests

drop-schema-query.test.ts: HANA supports dropSchema.cascade

dazhuangc committed:
feat(hana)!: HANA integration tests

describeTable.test.js: create table with comments on columns

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table-query.test.ts: column comments

dazhuangc committed:
feat(hana)!: HANA unit tests

drop-table-query.test.ts: IfExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

insert-query.test.ts: return ID workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

list-schemas-query.test.ts: SYS.SCHEMAS

dazhuangc committed:
feat(hana)!: HANA unit tests

list-tables-query.test.ts: SYS.TABLES

dazhuangc committed:
feat(hana)!: HANA unit tests

remove-column-query.test.ts: DROP ("column_name")

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table-query.test.ts: IF NOT EXISTS use SYS.TABLES
drop-table-query.test.ts: IF EXISTS use SYS.TABLES

dazhuangc committed:
feat(hana)!: HANA integration tests

list-tables.test.ts: use SYS.VIEWS instead of VIEWS

dazhuangc committed:
feat(hana)!: HANA unit tests

remove-index-query.test.ts: IfExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table-query.test.ts: ifNotExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

drop-table-query.test.ts: IfExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

remove-index-query.test.ts: IfExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

rename-table-query.test.ts: RENAME TABLE x TO y

dazhuangc committed:
feat(hana)!: HANA unit tests

select-query.test.ts: supports offset without limit

dazhuangc committed:
feat(hana)!: HANA unit tests

show-constraints-query.test.ts: SELECT FROM SYS.CONSTRAINTS/SYS.REFERENTIAL_CONSTRAINTS

dazhuangc committed:
feat(hana)!: HANA unit tests

show-indexes-query.test.ts: SELECT FROM SYS.INDEXES

dazhuangc committed:
feat(hana)!: HANA unit tests

start-transaction-query.test.ts: not supported

dazhuangc committed:
feat(hana)!: HANA unit tests

table-exists-query.test.ts: SELECT FROM SYS.TABLES

dazhuangc committed:
feat(hana)!: HANA unit tests

toggle-foreign-key-checks-query.test.ts: not supported

dazhuangc committed:
feat(hana)!: HANA unit tests

truncate-table-query.test.ts: TRUNCATE TABLE xx

dazhuangc committed:
feat(hana)!: HANA unit tests

update-query.test.ts: binds boolean values

dazhuangc committed:
feat(hana)!: HANA unit tests

version-query.test.ts: SELECT FROM SYS.M_SYSTEM_OVERVIEW

dazhuangc committed:
feat(hana)!: HANA unit tests

bulk-insert.test.ts: use HANA bulkInsertQuery()

dazhuangc committed:
feat(hana)!: HANA unit tests

bulk-insert-query.test.ts

dazhuangc committed:
feat(hana)!: HANA unit tests

schemas.test.ts: user name cannot be quoted in CREATE/DROP USER statement

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table.test.ts: ifNotExists workaround; uuid uses VARCHAR(36)

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table.test.ts: JSON uses NVARCHAR(5000)

dazhuangc committed:
feat(hana)!: HANA unit tests

drop-table.test.ts: ifExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

insert.test.ts: return ID workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

add-column.test.js: ALTER TABLE xx ADD (...)

dazhuangc committed:
feat(hana)!: HANA unit tests

change-column.test.js: properly generate alter queries

dazhuangc committed:
feat(hana)!: HANA unit tests

create-table.test.js: ifNotExists workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

index.test.js: addIndex - POJO field

dazhuangc committed:
feat(hana)!: HANA unit tests

insert.test.js: return ID workaround

dazhuangc committed:
feat(hana)!: HANA unit tests

order.test.js: supports random ordering

dazhuangc committed:
feat(hana)!: HANA unit tests

select.test.js: properly stringify IN values as per field definition

dazhuangc committed:
feat(hana)!: HANA unit tests

where.test.ts: Buffer
set the 'default' field of expectations

dazhuangc committed:
feat(hana)!: HANA unit tests

insert.test.ts: support parameter literal values:
 - firstName: literal('$firstName'),
 - firstName: literal('$1'),

dazhuangc committed:
feat(hana)!: HANA unit tests

insert-query.test.ts: support parameter literal values without options.bind:
 - lastName: literal('$lastName'),
 - lastName: literal('$1'),

dazhuangc committed:
feat(hana)!: HANA unit tests

index.test.js: type and using

dazhuangc committed:
feat(hana)!: HANA unit tests

select.test.js: hana does not support correlated subquery with ORDER BY

dazhuangc committed:
feat(hana)!: HANA unit tests

remove-index-query.test.ts: hana does not support cascade, concurrently

dazhuangc committed:
feat(hana)!: HANA integration tests

schemas.test.ts: should allow the user to specify indexes in options

dazhuangc committed:
feat(hana)!: HANA integration tests

cls.test.ts: Model.upsert
Although HANA does not support UPSERT

dazhuangc committed:
feat(hana)!: HANA integration tests

list-tables.test.ts: should not contain views

dazhuangc committed:
feat(hana)!: HANA dialect

remove unused import

dazhuangc committed:
feat(hana)!: HANA dialect

remove comment

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA unit tests

transaction.test.ts: hana supports.connectionTransactionMethods is true

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA unit tests

list-schemas-query.test.ts: filter schemas
list-tables-query.test.ts: filter schemas
set-isolation-level-query.test.ts: not support READ_UNCOMMITTED

dazhuangc committed:
feat(hana)!: HANA unit tests

list-schemas-query.test.ts: 'PUBLIC' is also a system defined schema
list-tables-query.test.ts: 'PUBLIC' is also a system defined schema

dazhuangc committed:
feat(hana)!: HANA unit tests

table-exists-query.test.ts: SYS.TABLES does not need quotes

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA unit tests

insert-query.test.ts: support parameter literal values without options.bind:
- lastName: literal('$lastName'),
- lastName: literal('$1'),

dazhuangc committed:
feat(hana)!: HANA dialect

HANA does not support bulkDefault

dazhuangc committed:
feat(hana)!: HANA dialect

HANA fromClause ' FROM DUMMY'

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA dialect

fix typo

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA dialect

clean code

dazhuangc committed:
feat(hana)!: HANA dialect

HANA connection does not support typeCast and timezone

dazhuangc committed:
feat(hana)!: HANA integration tests

connection-manager.test.js: should acquire a valid connection with connection ID

dazhuangc committed:
feat(hana)!: HANA dialect

HanaConnectionOptions.database is tenant database name

dazhuangc committed:
feat(hana)!: HANA dialect

remove console.log()

dazhuangc committed:
feat(hana)!: HANA dialect

remove commented code

dazhuangc committed:
feat(hana)!: HANA dialect

restore db2 and ibmi dependencies, which cause build failure

dazhuangc committed:
feat(hana)!: HANA integration tests

error.test.ts: handle different error message on HANA Cloud
HANA cloud: Error: failed on update or delete by foreign key constraint violation: failed on table 'USR_1IRYLIFZOR3BZD3LBOE9O810E:Users' with error:
HANA Express: foreign key constraint violation: TrexColumnUpdate failed on table '(.*):(.*)'

dazhuangc committed:
feat(hana)!: HANA integration tests

sequelize.test.js: handle invalid connection on HANA Cloud
1. HANA Cloud does not reject wrong port 9999, and a timeout will occur.
   So we should use host: 'bad-address'.
2. HANA Cloud error message contains 'Connection failed'
HANA Cloud: Connection failed (RTE:[89001] Cannot resolve host name 'bad-address' rc=-2: Name or service not known (bad-address:443))
HANA Express: Connection failed (RTE:[89001] Cannot resolve host name 'bad-address' rc=-2: Name or service not known (bad-address:39017))

dazhuangc committed:
feat(hana)!: HANA dialect

add HANA Cloud technical schema names

dazhuangc committed:
feat(hana)!: HANA dialect

parseConnectionUrl()

dazhuangc committed:
feat(hana)!: HANA dialect

remove commented duplicate properties in HanaConnectionOptions

dazhuangc committed:
feat(hana)!: HANA dialect

fix lint

dazhuangc committed:
feat(hana)!: HANA dialect

docker healthcheck

dazhuangc committed:
feat(hana)!: HANA dialect

update document in md files and package.json

dazhuangc committed:
feat(hana)!: HANA dialect

sscce-hana in package.json

dazhuangc committed:
feat(hana)!: HANA integration tests

sequelize.test.js: fix connection options property

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'hana-dialect-document' into hana-dialect-for-pr

dazhuangc committed:
feat(hana)!: HANA integration tests

select.test.js: restore tests after PR 17354 merged

dazhuangc committed:
feat(hana)!: HANA dialect

update yarn.lock

dazhuangc committed:
feat(hana)!: HANA integration tests

select.test.js: restore tests after PR 17354 merged

dazhuangc committed:
feat(hana)!: HANA dialect

fix prettier

dazhuangc commented:
Could someone help take a look at the failed check [CI / Lint code (pull_request)](https://github.com/sequelize/sequelize/actions/runs/13757470193/job/38467198861?pr=17758)? 
It is weird that prettier finished successfully with log `[prettier] All matched files use Prettier code style!`, but exited with code 2.
On my laptop, I can also get log `[prettier] All matched files use Prettier code style!`.

```
Run yarn test:format
[eslint] 
[eslint] > test:format:eslint
[eslint] > eslint . --quiet --report-unused-disable-directives
[eslint] 
[prettier] 
[prettier] > test:format:prettier
[prettier] > prettier --check .
[prettier] 
[prettier] Checking formatting...
[prettier] 
Error: [prettier] [error] Unable to read file "install-build-node-20.tar":
Error: [prettier] [error] Invalid string length
[prettier] All matched files use Prettier code style!
[prettier] npm run test:format:prettier exited with code 2
[eslint] npm run test:format:eslint exited with code 0
Error: Process completed with exit code 1.
```
I checked the log, but cannot figure out why CI / Lint code failed.
Has anyone encountered this problem?

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana)!: HANA dialect

meta: update dependency

dazhuangc committed:
feat(hana): HANA dialect

add hana in scopes in pr-title.yml

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

fix CI / Lint code: add install-build-node-20.tar in .prettierignore

dazhuangc committed:
feat(hana)!: HANA dialect

update version in package.json

dazhuangc committed:
feat(hana): HANA dialect

add test-unit-hana in unit-test in ci.yml

dazhuangc committed:
feat(hana): HANA dialect

add dialect.test.ts in src

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

remove dev/hana/oldest/

dazhuangc committed:
feat(hana): HANA dialect

remove hostname in dev/hana/latest/docker-compose.yml

dazhuangc committed:
feat(hana): HANA dialect

remove start-hana-oldest

dazhuangc committed:
feat(hana): HANA dialect

add test-hana (latest only) in jobs in ci-title.yml

dazhuangc committed:
feat(hana): HANA dialect

add test-hana (latest only) in jobs in ci-title.yml

dazhuangc committed:
feat(hana): HANA dialect

fix prettier in ci-title.yml

dazhuangc committed:
feat(hana): HANA dialect

fix prettier in ci.yml

dazhuangc committed:
feat(hana): HANA dialect

1. before start docker, add /data/hxe/ and /data/hxe/hxepassword.json and set permission
  see https://hub.docker.com/r/saplabs/hanaexpress
2. use 240s timeout in hana own wait-until-healthy.sh
  usually it takes ~200s to get started and healthy

dazhuangc committed:
feat(hana): HANA dialect

add executable permission to dev/hana/wait-until-healthy.sh
mode: 644->755

dazhuangc commented:
Tried adding `install-build-node-20.tar` in `.prettierignore`, then the `CI / Lint code` can pass (commit https://github.com/sequelize/sequelize/pull/17802/commits/e433940fe82bb615b26abf96191bd4511ce988e7 in PR https://github.com/sequelize/sequelize/pull/17802). Not sure why.
Should `install-build-node-20.tar` be added into .prettierignore in current PR?

WikiRik commented:
Preferably not, but if that works it's fine for now

dazhuangc committed:
feat(hana): HANA dialect

add test-hana in 4. Running tests in CONTRIBUTING.md

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

upsert.test.ts: remove hana assert because hana does not support upserts

dazhuangc committed:
feat(hana): HANA dialect

use uppercase AS keyword

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

fix lint after merge

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc commented:
The CI fails with error:
```
Exception during run: Error: The "native" option was specified, but the "pg-native" module is not installed. You must install it to use the native bindings.
```
Even if no changes are made: https://github.com/sequelize/sequelize/pull/17849

Are there any changes related to the CI issue?

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc commented:
> The CI fails with error:
> 
> ```
> Exception during run: Error: The "native" option was specified, but the "pg-native" module is not installed. You must install it to use the native bindings.
> ```
> 
> Even if no changes are made: #17849
> 
> Are there any changes related to the CI issue?

This issue does not occur after merging main branch at 87bfd061f1734d3a0b476295bf0cae6e3e204b22.

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc commented:
Added documentation update PR: https://github.com/sequelize/website/pull/799

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
meta: update dependency mocha to v11.2.2

commit 700de2c3c48208945bdda4544124a60cbd6698e5

dazhuangc committed:
meta: update dependency mocha to v11.2.2

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
meta: update yarn.lock

dazhuangc commented:
Hi @WikiRik,
I have been keeping this PR up to date, and checked if current PR has the same issues as commented in https://github.com/sequelize/sequelize/pull/16394.
And added documentation update PR.

Is there anything I can do to process this PR?

dazhuangc commented:
> Unfortunately you just need to wait until we get the chance to properly review this. Personally I think it's realistic that we do not include this in the initial v7 release and add this in a later minor version once we have discovered issues others run into

Hi Rik, do you have a planned time for the initial version 7 release?

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
meta: update dependency mocha to v11.5.0

dazhuangc committed:
meta: update yarn.lock

dazhuangc commented:
Hi @WikiRik , 
In https://github.com/sequelize/sequelize/discussions/16982 a dialect plugin system is mentioned. Is it still planned in Sequelize v8?

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
meta: update dependency mocha to v11.6.0

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
meta: update dependency mocha to v11.7.1

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

WikiRik commented:
I'll merge https://github.com/sequelize/sequelize/pull/17954 in a bit to deal with the prettier issue we're running into

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'sequelize:main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

query-generator.internal.ts: add SYS_DATABASES as technical schema

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

cls.test.ts: fix bad merge

dazhuangc committed:
Merge branch 'hana-dialect-for-pr' of https://github.com/dazhuangc/sequelize into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

query-generator.js: fix missing bindParam()
query-generator-typescript.internal.ts: add getRandomFloatFunctionCall()

dazhuangc committed:
meta: update dependency mocha to v11.7.5

dazhuangc committed:
feat(hana): HANA dialect

create-schema-query.test.ts: fix lint

dazhuangc committed:
feat(hana): HANA dialect

fix unit tests

dazhuangc committed:
feat(hana): HANA dialect

support.ts: add hana dialect

dazhuangc committed:
feat(hana): HANA dialect

test: fix DecrementedDepth

dazhuangc committed:
feat(hana): HANA dialect

test: fix DecrementedDepth

dazhuangc committed:
feat(hana): HANA dialect

ci.yml: update steps version

dazhuangc committed:
feat(hana): HANA dialect

query.test.js: add hana in test

dazhuangc committed:
feat(hana): HANA dialect

query-generator-typescript.internal.ts: test 'should allow the user to specify indexes in options'

dazhuangc committed:
feat(hana): HANA dialect

query-generator-typescript.internal.ts: test 'should allow the user to specify indexes in options'

dazhuangc committed:
feat(hana): HANA dialect

query-generator-typescript.internal.ts: test 'should allow the user to specify indexes in options'

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

packages/hana/src/dialect.ts: remove getDefaultPort()

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
Revert "feat(hana): HANA dialect"

This reverts commit 85b8f0dea738824e8a42d2267ade3ee8e2ec0428.

dazhuangc committed:
Merge branch 'main' into hana-dialect-for-pr

dazhuangc committed:
feat(hana): HANA dialect

fix unit tests: temporal-types.test.ts, where.test.ts
